src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java

```diff
@@ -0,0 +1,286 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Retrieves historical confirmed case data
+ * starting from 1-22-20. Takes coordinates
+ * in request and returns closest report
+ */
+@WebServlet("/timereport")
+public class OverTimeCasesServlet extends HttpServlet {
+  private HashMap<LocLatLng, ArrayList<Integer>> usTimeReports;
+  private HashMap<LocLatLng, ArrayList<Integer>> globalTimeReports;
+  private ArrayList<Integer> worldCases;
+  private ArrayList<String> dates;
+  public static final String CTYPE = "application/json"; // HttpServletResponse content type
+  public static final String ENCODING = "UTF-8"; // HttpServletResponse character encoding
+
+  /**
+   * Builds report hashmaps for US counties and international countires
+   */
+  @Override
+  public void init() {
+    // Build US hashmap
+    usTimeReports = new HashMap<LocLatLng, ArrayList<Integer>>();
+    Scanner usScanner = connectToData("US");
+    fillDataMap(usScanner, usTimeReports, 7, 6, 5, 9);
+
+    // Build international hashmap
+    globalTimeReports = new HashMap<LocLatLng, ArrayList<Integer>>();
+    Scanner globalScanner = connectToData("global");
+    fillDataMap(globalScanner, globalTimeReports, 0, 0, 0, 0);
+  }
+
+  /**
+   * Returns history of confirmed cases for the location specificed by cooridnates
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setCharacterEncoding(ENCODING);
+    response.setContentType(CTYPE);
+
+    // Get coordinates from request
+    double lat = Double.parseDouble(getRequestParameterOrDefault(request, "lat", "0.0"));
+    double lng = Double.parseDouble(getRequestParameterOrDefault(request, "lng", "0.0"));
+
+    // Coordinates 0 0 are located in the atlantic ocean, will be used to request worldwide cases
+    if (lat == 0.0 && lng == 0.0) {
+      LocationCases toReturn = new LocationCases("Worldwide", worldCases, dates);
+      Gson gson = new Gson();
+      String timeReportJson = gson.toJson(toReturn);
+      response.getWriter().println(timeReportJson);
+      return;
+    }
```

Comment:
another style thing (i'm sorry). it's usually not preferred to return unless you actually need to return a value. In this case (based on my understanding), everything from line 78 (to 114) should be in a massive else branch, because this method is doing x if lat and long = 0, else it will do whatever is below it. I think the idea is that else makes it more readable.